backend/s3/s3.go

```diff
@@ -2349,7 +2376,13 @@ func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, op
 		req.ContentMD5 = &md5sum
 	}
 	if o.fs.opt.ServerSideEncryption != "" {
-		req.ServerSideEncryption = &o.fs.opt.ServerSideEncryption
```

Comment:
Can you justify this change? is `req.ServerSideEncryption` wrong here or deprecated in favor of `req.SSECustomerAlgorithm` or something else?

Reply:
Yeah my bad, I just misinterpreted documentation. Looks like 'ServerSideEncryption' is still valid if you're not using SSE-C.